test/libsolidity/InlineAssembly.cpp

```diff
@@ -223,6 +223,48 @@ BOOST_AUTO_TEST_CASE(variable_use_before_decl)
 	CHECK_PARSE_ERROR("{ let x := mul(2, x) }", DeclarationError, "Variable x used before it was declared.");
 }
 
+BOOST_AUTO_TEST_CASE(switch_statement)
+{
+	BOOST_CHECK(successParse("{ switch 42 default {} }"));
+	BOOST_CHECK(successParse("{ switch 42 case 1 {} }"));
+	BOOST_CHECK(successParse("{ switch 42 case 1 {} case 2 {} }"));
+	BOOST_CHECK(successParse("{ switch 42 case 1 {} default {} }"));
+	BOOST_CHECK(successParse("{ switch 42 case 1 {} case 2 {} default {} }"));
+	BOOST_CHECK(successParse("{ switch mul(1, 2) case 1 {} case 2 {} default {} }"));
+	BOOST_CHECK(successParse("{ function f() -> x {} switch f() case 1 {} case 2 {} default {} }"));
+}
+
+BOOST_AUTO_TEST_CASE(switch_no_cases)
+{
+	CHECK_PARSE_ERROR("{ switch 42 }", ParserError, "Switch statement without any cases.");
+}
+
+BOOST_AUTO_TEST_CASE(switch_duplicate_case)
+{
+	CHECK_PARSE_ERROR("{ switch 42 case 1 {} case 1 {} default {} }", DeclarationError, "Duplicate case defined");
+}
+
+BOOST_AUTO_TEST_CASE(switch_invalid_expression)
+{
+	CHECK_PARSE_ERROR("{ switch {} default {} }", ParserError, "Expected elementary inline assembly operation.");
+	CHECK_PARSE_ERROR("{ 1 2 switch mul default {} }", ParserError, "Instructions are not supported as expressions for switch.");
+}
+
+BOOST_AUTO_TEST_CASE(switch_default_before)
+{
+	CHECK_PARSE_ERROR("{ switch 42 default {} case 1 {} }", ParserError, "Expected elementary inline assembly operation.");
```

Comment:
There could a peek in the parser for `Default` and `Case` tokens to provide a more user friendly error message (i.e. "case not allowed after default case" and "only one default case allowed")